codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -560,11 +560,15 @@ public ChannelFuture goAway(final ChannelHandlerContext ctx, final int lastStrea
             future.addListener(new GenericFutureListener<ChannelFuture>() {
                 @Override
                 public void operationComplete(ChannelFuture future) throws Exception {
-                    if (!future.isSuccess()) {
+                    if (future.isSuccess()) {
+                        if (errorCode != NO_ERROR.code()) {
+                            ctx.close();
+                        }
+                    } else {
                         String msg = format("Sending GOAWAY failed: lastStreamId '%d', errorCode '%d', " +
```

Comment:
@Scottmitch wrap this in an if statement as we not need to do the format stuff if error logger is not enabled:

```java
if (logger.isErrorEnabled()) {
    String msg = format(....);
    logger.error(msg);
}
ctx.close();
....
```